src/java.desktop/share/classes/javax/imageio/ImageIO.java

```diff
@@ -1464,6 +1464,8 @@ public static BufferedImage read(ImageInputStream stream)
         BufferedImage bi;
         try (stream) {
             bi = reader.read(0, param);
+        } catch (RuntimeException e) {
```

Comment:
Is that necessary to catch all possible runtime exceptions? Probably we should catch only the exceptions thrown by the read method? I think it should be "IllegalArgumentException" only, the IllegalStateException/IndexOutOfBoundsException should not be thrown here, and the IOException is fine.

Reply:
It is a wrapper that is intended to hide lots of details - such as this.
A sophisticated app which wants to use ImageReader.read() should be ready to handle those
An app that just wants a quick and easy API with no random exceptions wants ImageIO.read().
And as I said the exception is wrapped so it can still be inspected.

Reply:
> A sophisticated app which wants to use ImageReader.read() should be ready to handle those

The current method differs from the "read" only by the "IllegalArgumentException", all other exceptions we will handle here will mean a bug in the read method, which also can be used by the app.

I do not understand why the app should be ready for unexpected/unspecified exceptions in case of "read" method and could not be ready in this one.